RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractEdge.java

```diff
@@ -0,0 +1,197 @@
+            File datFile = new File(temps);
+            if (context.dataSourceIngestIsCancelled()) {
+                break;
+            }
+            try {
+                ContentUtils.writeToFile(indexFile, datFile, context::dataSourceIngestIsCancelled);
+            } catch (IOException e) {
+                logger.log(Level.WARNING, "Error while trying to write index.dat file " + datFile.getAbsolutePath(), e); //NON-NLS
+                this.addErrorMessage(
+                        NbBundle.getMessage(this.getClass(), "ExtractEdge.process.errMsg.errWriteFile", this.getName(),
+                                datFile.getAbsolutePath()));
+                continue;
+            }
+            
+            File resultsDir = new File(moduleTempResultsDir + Integer.toString((int) indexFile.getId()));
+            resultsDir.mkdirs();
+            executeDumper(esedumper, datFile.getAbsolutePath(), "webcache", resultsDir.getAbsolutePath());
+             
+            this.getHistory(); // Not implemented yet
+            this.getCookie(); // Not implemented yet
+            this.getDownload(); // Not implemented yet
+            
+             datFile.delete();
+             resultsDir.delete();
+         }   
+    }
+    
+    /**
+     * Query for history databases and add artifacts
+     */
+    private void getHistory() {
+        
+    }
+    
+    /**
+     * Search for bookmark files and make artifacts.
+     */
+    private void getBookmark() {
+        
+    }
+    
+    /**
+     * Queries for cookie files and adds artifacts
+     */
+    private void getCookie() {
+        
+    }
+    
+    /**
+     * Queries for download files and adds artifacts
+     */
+    private void getDownload() {
+        
+    }
+    
+    private boolean executeDumper(String dumperPath, String inputFilePath, String inputFilePrefix, String outputDir){
+        final String outputFileFullPath = outputDir + File.separator + inputFilePrefix + ".txt";
+        final String errFileFullPath = outputDir + File.separator + inputFilePrefix + ".err"; //NON-NLS
+        logger.log(Level.INFO, "Writing ESEDatabaseViewer results to: {0}", outputDir); //NON-NLS   
+        
+        List<String> commandLine = new ArrayList<>();
+        commandLine.add(dumperPath);
+        commandLine.add("/table");
+        commandLine.add(inputFilePath);
+         commandLine.add("*");
+        commandLine.add("/scomma");
+        commandLine.add(outputDir + "\\" + inputFilePrefix + "_*.csv");
+         
+        ProcessBuilder processBuilder = new ProcessBuilder(commandLine);
+        processBuilder.redirectOutput(new File(outputFileFullPath));
+        processBuilder.redirectError(new File(errFileFullPath));
+
+        try{
+            ExecUtil.execute(processBuilder, new DataSourceIngestModuleProcessTerminator(context));
+        }catch(IOException ex){
+            logger.log(Level.SEVERE, "Unable to execute ESEDatabaseView to process Edge file." , ex); //NON-NLS
```

Comment:
Which Edge file?

Personally, I think it is better to throw exceptions than have boolean or error code return values; the general policy is lower level code throws exceptions for handling by higher level code. I would just propagate the IOException here - it is cleaner than having multiple return statements fro true/false.

Also, be sure to format (ALT+SHIFT+F) before check in. For example, the formatter will add space between the reserved words and the "curly" braces of this try-catch.

Reply:
I agree with that too, but I noticed that the examples I was looking at didn't.   It will make me feel better to change it.  :-)

Reply:
Me, too! :)